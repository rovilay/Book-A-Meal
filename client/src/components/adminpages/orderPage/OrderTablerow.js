/* eslint no-nested-ternary: 0 */
import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';
import ReactTooltip from 'react-tooltip';

const OrderTableRow = ({
  item,
  mealsUrl,
  actions,
  showDetails
}) => (
  <div className="row">
    {
      Object.keys(item).map((key, i) => (
        (
          (key === 'orderId' || key === 'Meal')
            ?
            (
              <p
                key={i}
                className="row-item meal-title"
                data-tip={item[key]}
              >
                {item[key]}
              </p>
            )
            :
            (
              <p
                key={i}
                className="row-item"
              >
                {item[key]}
              </p>
            )
        )
      ))
    }

    {
      (actions)
      &&
      (
      <p
        className="row-item actions"
      >

        {
          (actions.info)
          &&
          (
            <button
              className="btn-3 box-shadow"
              onClick={() => {
                showDetails(mealsUrl);
              }}
            >
              <FontAwesome
                name="info-circle"
                size="2x"
                className="trash info-circle"
              />
            </button>
          )
        }
      </p>
      )
    }

    <ReactTooltip className="mytooltip" />
  </div>
);

OrderTableRow.defaultProps = {
  showDetails: undefined,
  mealsUrl: undefined,
  actions: undefined,
};

OrderTableRow.propTypes = {
  item: PropTypes.object.isRequired,
  showDetails: PropTypes.func,
  mealsUrl: PropTypes.string,
  actions: PropTypes.object
};

export default OrderTableRow;
