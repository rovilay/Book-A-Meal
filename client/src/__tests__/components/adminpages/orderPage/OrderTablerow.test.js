import React from 'react';
import { shallow } from 'enzyme';
import OrderTablerow from '../../../../components/adminpages/orderPage/OrderTablerow';
import { orderDetails } from '../../../__mockData__/orderMock';


const props = {
  item: {
    ...orderDetails
  },
  showDetails: jest.fn(),
  mealsUrl: 'orders/1/meals',
  actions: {
    delete: true,
    info: true,
    edit: true
  }
};

describe('OrderTablerow component test', () => {
  const setup = () => shallow(<OrderTablerow {...props} />);


  it('should render `OrderTablerow` correctly', (done) => {
    const wrapper = setup();

    wrapper.find('.btn-3').simulate('click');
    expect(wrapper).toMatchSnapshot();
    expect(props.showDetails).toHaveBeenCalled();

    done();
  });
});
