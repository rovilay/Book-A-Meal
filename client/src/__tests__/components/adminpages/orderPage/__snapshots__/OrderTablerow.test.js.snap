// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderTablerow component test should render \`OrderTablerow\` correctly 1`] = `
<div
  className="row"
>
  <p
    className="row-item meal-title"
    data-tip={6}
    key="0"
  >
    6
  </p>
  <p
    className="row-item"
    key="1"
  >
    isuti rd
  </p>
  <p
    className="row-item"
    key="2"
  >
    <Component />
    <Component />
  </p>
  <p
    className="row-item actions"
  >
    <button
      className="btn-3 box-shadow"
      onClick={[Function]}
    >
      <FontAwesome
        className="trash info-circle"
        name="info-circle"
        size="2x"
      />
    </button>
  </p>
  <ReactTooltip
    className="mytooltip"
    insecure={true}
    resizeHide={true}
    wrapper="div"
  />
</div>
`;
